### PR TITLE
EA Forum Sidebar Redesign

### PR DESCRIPTION
**Pull request by oetherington**
This PR is the post-subforum redesign of the sidebar for the EA Forum.

For the most part I was able to just follow the Figma design, but there was one issue. The heading "Topics" itself is a link to the EA Wiki (at least at the moment) so the extra "EA Wiki" link was redundant. For this reason, I've left out the seperate link and also removed the "Subforums" heading as well since there's nothing else left in this dropdown.

I'm not 100% sure this is the correct solution, but it seemed like the most straightforward - it could be that we want to do something else to make both the wiki and subforums more obvious though. If you'd prefer to do something else then either feel free to make the changes whilst I'm away tomorrow, or if it's not so urgent I can change it on Monday.

I also left in the links under the divider as I assumed these were just left out of the design for the sake of doing less Figma work rather than because we actually want to remove them. If that's not the case then removing them should just be a matter of deleting some stuff from `menuTabs.ts`.

--- Update from @Will-Howard 2022-12-15:

I've added "EA Wiki" and "(see all)"

Screenshot 2022-12-15 at 15 25 28
Screenshot 2022-12-15 at 15 26 26
(second image is logged in, hence why "Take Action" isn't there)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203435749850299) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Will-Howard** [comment]
Thanks Ollie! I'll take this over to finish it off. The main thing is making "Topics" not be a link and adding "EA wiki" instead, it wasn't obvious that was what we wanted to do from the design

File: packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx [hunk 1]
```diff
@@ -19,58 +21,90 @@ const styles = ((theme: ThemeType): JssStyles => ({
...
+                  <TabNavigationSubItem className={classes.subItem}>
+                    <div className={classes.indented}>{startCase(subforum.name)}</div>
+                  </TabNavigationSubItem>
+                </MenuItemUntyped>
+              ))}
+              <MenuItemUntyped
+                key="all-subforums"
+                onClick={onClick}
+                component={Link}
+                to={"/search?contentType=Tags&query=&toggle%5BisSubforum%5D=true&page=1"}
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx:99]: links to search page with "Has Subforum" selected

File: packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx [hunk 2]
```diff
@@ -271,42 +277,56 @@ export const menuTabs: ForumOptions<Array<MenuTab>> = {
...
+      },
     }, {
-      id: 'eventsList',
-      customComponentName: "EventsList",
+      id: 'takeAction',
+      title: 'Take Action',
+      link: `/${taggingNamePluralSetting.get()}/take-action`,
+      iconComponent: PlaylistAddCheck,
+      tooltip: "Opportunities to get involved with impactful work",
+      loggedOutOnly: true
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx:329]: I reordered this just to keep all the collapsible ones together

File: packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx [hunk 3]
```diff
@@ -19,58 +21,90 @@ const styles = ((theme: ThemeType): JssStyles => ({
...
+                classes={{ root: classes.menuItem }}
+              >
+                <TabNavigationSubItem className={classes.subItem}>
+                  <div className={classes.indented}>(see all)</div>
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx:103]: I don't love how this looks, but it also looks a bit strange without this when there is only one subforum (which will be the case for logged out users)

File: packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx (hunk 2 above)
Comment (s-cheng) [reply packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx:329]: Hmm I would put this higher up - at least above Events, but maybe even above Topics (if you want to keep the collapsible menu items together).

Also, would it be easy to add the sub-menu items back in for Community, now that we can just make it collapsed by default? (It was Local Groups, Online Groups, and Community Members, pointing to the three tabs on that page.) If not, then I can do that in a separate PR

**Comment by s-cheng** [comment]
Why not link the "Topics" menu item to the wiki?

**Comment by s-cheng** [comment]
I like the expand/collapse animation! Though I think it would be ideal if all menu items were links, in order to be consistent.

I'm kind of confused though, because I thought the goal of redesigning the sidebar was to clean it up, but this seems longer than the existing sidebar? I also feel like we shouldn't have any of the menu items expanded by default, since that makes the fact that they are collapsible pretty irrelevant since I don't think users would click to collapse them.

Here is what I'm imagining:
1. All the menu items are collapsed so that the featured banner has a chance to appear above the fold
2. "Take Action" is higher than "Events" since it's more important - perhaps even above "Topics Wiki" as I put here (I also changed the icon since it looked too similar to the "All Posts" icon)
3. "Subforums" is separate from "Topics Wiki", since I feel that from a user perspective, it's confusing to combine them (at least in their current state)

Perhaps I'm misunderstanding the purpose of this though? I will ask in slack, and see what Sharang thinks. (I haven't reviewed the code itself yet - will do so once we have clarity on this.)

Screen Shot 2022-12-16 at 12 13 25 AM

File: packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx (hunk 2 above)
Comment (Will-Howard) [reply packages/lesswrong/components/common/TabNavigationMenu/menuTabs.tsx:329]: Yep, I can add those back in

**Comment by Will-Howard** [comment]
> I like the expand/collapse animation! Though I think it would be ideal if all menu items were links, in order to be consistent.
> 
> I'm kind of confused though, because I thought the goal of redesigning the sidebar was to clean it up, but this seems longer than the existing sidebar? I also feel like we shouldn't have any of the menu items expanded by default, since that makes the fact that they are collapsible pretty irrelevant since I don't think users would click to collapse them.
> 
> Here is what I'm imagining:
> 
> 1. All the menu items are collapsed so that the featured banner has a chance to appear above the fold
> 2. "Take Action" is higher than "Events" since it's more important - perhaps even above "Topics Wiki" as I put here (I also changed the icon since it looked too similar to the "All Posts" icon)
> 3. "Subforums" is separate from "Topics Wiki", since I feel that from a user perspective, it's confusing to combine them (at least in their current state)
> 
> Perhaps I'm misunderstanding the purpose of this though? I will ask in slack, and see what Sharang thinks. (I haven't reviewed the code itself yet - will do so once we have clarity on this.)
> 
> Screen Shot 2022-12-16 at 12 13 25 AM

I agree the goals of this have gotten a bit muddled. The goal for subforums specifically was to make it clearer that subforums are a special kind of topic, which is why I moved the link from "Topics Wiki" down to the "EA Wiki" sub item. The thing I'm trying to convey is "there are these things called topics, one of the things you might want to look at is a wiki of all the topics, you might also want to look at one of these special topics called subforums". Then there was also the goal of cleaning the sidebar up which got rolled into this

I agree that it doesn't add much to make the items collapsible while keeping them open by default. Overall I prefer how it looks with them all collapsed, but:
 - I would like to keep subscribed subforums accessible in one click from the front page, so I would prefer that the subforum/topics item wasn't collapsed
 - I would like to keep subforums under Topics, although I'd be willing to let this go seeing as the pages currently look quite different

Does this change your opinion at all?

**Pull request closed without merging**